### PR TITLE
feat(flags): Store options changes in the audit log

### PR DESCRIPTION
**Pull request by cmanallen**
When an option is added, updated, or removed record that action in the audit log.

### CODE REVIEW

File: src/sentry/flags/providers.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+def handle_flag_pole_event_internal(items: list[FlagAuditLogItem], organization_id: int) -> None:
+    write(
+        [
+            {
+                "action": ACTION_MAP[item["action"]],
+                "created_at": item["created_at"],
+                "created_by": item["created_by"],
+                "created_by_type": CREATED_BY_TYPE_MAP["name"],
```
Comment (michellewzhang) [inline src/sentry/flags/providers.py:78]: `"created_by_type": CREATED_BY_TYPE_MAP["name"],`

--> why did it change from "email" to "name"?

File: src/sentry/runner/commands/presenters/audit_log_presenter.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                "created_at": datetime.now(tz=timezone.utc),
+                "created_by": "internal",
+                "flag": flag,
+                "tags": tags,
+            }
+            for flag, action, tags in itertools.chain(
+                ((flag, "created", {"value": v}) for flag, v in self.set_options),
+                ((flag, "deleted", {}) for flag in self.unset_options),
+                ((flag, "updated", {"value": v}) for flag, _, v in self.updated_options),
+                ((flag, "updated", {}) for flag, _ in self.drifted_options),
```
Comment (michellewzhang) [inline src/sentry/runner/commands/presenters/audit_log_presenter.py:43]: what does a drifted option mean?

File: src/sentry/flags/providers.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/flags/providers.py:78]: We lost the email information moving to this location.

File: src/sentry/runner/commands/presenters/audit_log_presenter.py (hunk 2 above)
Comment (cmanallen) [reply src/sentry/runner/commands/presenters/audit_log_presenter.py:43]: Internal lingo.  Basically the value was updated but the existing value in the db was unexpected.

Comment (michellewzhang) [reply src/sentry/runner/commands/presenters/audit_log_presenter.py:43]: we don't care about the new updated value in this case?

Comment (cmanallen) [reply src/sentry/runner/commands/presenters/audit_log_presenter.py:43]: We still do.  The value is changed in the db.  Its just that the old value was not what we expected (someone manually changed it).

Comment (cmanallen) [reply src/sentry/runner/commands/presenters/audit_log_presenter.py:43]: Oh I see the tags.  The drifted options don't contain the new value which is why it was omitted.  This is a legacy event any way.  I could probably remove it.

**Review by michellewzhang — APPROVED** [review]

**Pull request merged**